pkg/executor/nginx/nginx.go

```diff
@@ -197,3 +205,89 @@ func whitelistOverrides(overrides map[string]string) error {
 	}
 	return nil
 }
+
+func (m nginxLocations) sort() []string {
+	index := []string{}
+	for k := range m {
+		index = append(index, k)
+	}
+	sort.Strings(index)
+	return index
+}
+
+func (m headers) sort() []string {
+	index := []string{}
+	for k := range m {
+		index = append(index, k)
+	}
+	sort.Strings(index)
+	return index
+}
+
+func nginxLocationsMapToString(m nginxLocations, documentRoot string, path string) string {
+	sumLocations := ""
+	indentN1 := strings.Repeat(" ", 8)
+	indentN2 := strings.Repeat(" ", 12)
+
+	for _, k := range m.sort() {
+		v := m[k]
+		singleLocation := fmt.Sprintf("%slocation %s%s {\n", indentN1, path, k)
+		singleLocation = fmt.Sprintf("%s%sroot %s;\n", singleLocation, indentN2, documentRoot)
+		gZipUse := strings.TrimSpace(v.Gzip.Use)
+		gZipVary := strings.TrimSpace(v.Gzip.Vary)
+		if gZipUse == "on" {
+			singleLocation = fmt.Sprintf("%s%sgzip on;\n", singleLocation, indentN2)
+			if v.Gzip.MinLength > 0 {
+				singleLocation = fmt.Sprintf("%s%sgzip_min_length %d;\n", singleLocation, indentN2, v.Gzip.MinLength)
+			}
+			if gZipVary != "" {
+				singleLocation = fmt.Sprintf("%s%sgzip_vary %s;\n", singleLocation, indentN2, gZipVary)
+			}
+			if v.Gzip.Proxied != "" {
+				singleLocation = fmt.Sprintf("%s%sgzip_proxied %s;\n", singleLocation, indentN2, v.Gzip.Proxied)
+			}
+			if v.Gzip.Types != "" {
+				singleLocation = fmt.Sprintf("%s%sgzip_types %s;\n", singleLocation, indentN2, v.Gzip.Types)
+			}
+			if v.Gzip.Disable != "" {
+				singleLocation = fmt.Sprintf("%s%sgzip_disable \"%s\";\n", singleLocation, indentN2, v.Gzip.Disable)
+			}
+		} else if gZipUse == "off" {
+			singleLocation = fmt.Sprintf("%s%sgzip off;\n", singleLocation, indentN2)
+		}
+
+		for _, k2 := range v.Headers.sort() {
+			singleLocation = fmt.Sprintf("%s%sadd_header %s \"%s\";\n", singleLocation, indentN2, k2, v.Headers[k2])
+		}
+
+		singleLocation = fmt.Sprintf("%s%s}\n", singleLocation, indentN1)
+		sumLocations = sumLocations + singleLocation
+	}
+	return sumLocations
+}
+
+func nginxGzipMapToString(gzip nginxGzip) string {
```

Comment:
Kan du prøve å redusere kode dupliseringen i disse to metodene ved å benytte bedre abstraksjoner